### PR TITLE
LibJS: Separate validation of roundingIncrement option

### PR DESCRIPTION
**Pull request by dlarocque**
This is an editorial change in the temporal spec.

See: https://github.com/tc39/proposal-temporal/commit/712c449

Ticks off a box in https://github.com/SerenityOS/serenity/issues/15525

### CODE REVIEW

**Review by shannonbooth — CHANGES_REQUESTED** [review]
I've run this MR on [libjs-test262](https://github.com/SerenityOS/libjs-test262). Unfortunately it looks like the test `test/built-ins/Temporal/PlainTime/prototype/round/roundingincrement-invalid.js` failed after these changes (it previously passed).

One of the example assertion failures is:
```js
const plainTime = Temporal.PlainTime.from("08:22:36.123456789");
assert.throws(RangeError, () => plainTime.round({ smallestUnit: "minutes", roundingIncrement: 60 }));
```

**Comment by dlarocque** [comment]
> I've run this MR on [libjs-test262](https://github.com/SerenityOS/libjs-test262). Unfortunately it looks like the test `test/built-ins/Temporal/PlainTime/prototype/round/roundingincrement-invalid.js` failed after these changes (it previously passed).
> 
> One of the example assertion failures is:
> 
> ```js
> const plainTime = Temporal.PlainTime.from("08:22:36.123456789");
> assert.throws(RangeError, () => plainTime.round({ smallestUnit: "minutes", roundingIncrement: 60 }));
> ```

Seems like I passed the wrong value of `inclusive` to one of the calls to `validate_temporal_rounding_increment`- this test should be passing now :+1: 

Aside: Is there a convenient way to check which tests fail that previously passed?

**Comment by shannonbooth** [comment]
> > I've run this MR on [libjs-test262](https://github.com/SerenityOS/libjs-test262). Unfortunately it looks like the test `test/built-ins/Temporal/PlainTime/prototype/round/roundingincrement-invalid.js` failed after these changes (it previously passed).
> > One of the example assertion failures is:
> > ```js
> > const plainTime = Temporal.PlainTime.from("08:22:36.123456789");
> > assert.throws(RangeError, () => plainTime.round({ smallestUnit: "minutes", roundingIncrement: 60 }));
> > ```
> 
> Seems like I passed the wrong value of `inclusive` to one of the calls to `validate_temporal_rounding_increment`- this test should be passing now 👍
> 
> Aside: Is there a convenient way to check which tests fail that previously passed?

Nice! Could you please add a local `test-js` test to cover the regression?


In terms of getting the test results - first, you can establish a baseline (before any changes):

```
python3 main.py --libjs-test262-runner ./Build/bin/test262-runner --test262-root ./test262/ --per-file old
```

Then rebuild with your changes, get latest results:

```
python3 main.py --libjs-test262-runner ./Build/bin/test262-runner --test262-root ./test262/ --per-file new
```

Diff the results:
```
./per_file_result_diff.py -o old -n new
```

**Comment by dlarocque** [comment]
> Nice! Could you please add a local `test-js` test to cover the regression?
> 
> In terms of getting the test results - first, you can establish a baseline (before any changes):
> 
> ```
> python3 main.py --libjs-test262-runner ./Build/bin/test262-runner --test262-root ./test262/ --per-file old
> ```
> 
> Then rebuild with your changes, get latest results:
> 
> ```
> python3 main.py --libjs-test262-runner ./Build/bin/test262-runner --test262-root ./test262/ --per-file new
> ```
> 
> Diff the results:
> 
> ```
> ./per_file_result_diff.py -o old -n new
> ```

Thanks! I'll use that before submitting the PR next time :sweat_smile: 

I've added some tests to cover the regression

File: Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp [hunk 1]
```diff
@@ -274,48 +274,61 @@ ThrowCompletionOr<String> to_show_offset_option(VM& vm, Object const& normalized
...
+    }
 
-    // 7. Set increment to floor(ℝ(increment)).
+    // 5. Set increment to floor(ℝ(increment)).
     auto floored_increment = static_cast<u64>(increment);
 
-    // 8. If dividend is not undefined and dividend modulo increment ≠ 0, then
-    if (dividend.has_value() && static_cast<u64>(*dividend) % floored_increment != 0)
+    // 6. If dividend modulo increment is not zero, then
+    if (static_cast<u64>(dividend) % floored_increment != 0) {
```
[the remark this answers is not in the document]
Comment (dlarocque) [reply Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp]: Updated 👍

**Review by shannonbooth — APPROVED** [review]

**Pull request merged**